### PR TITLE
Update changelog for 1.9.0-rc1

### PR DESCRIPTION
**Pull request by kalafut**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,93 +1,175 @@
...
+  been removed. [[GH-12888](https://github.com/hashicorp/vault/pull/12888)]
+* go: Update go version to 1.17.2
+* secrets/ssh: Roles with empty allowed_extensions will now forbid end-users
+  specifying extensions when requesting ssh key signing. Update roles setting
+  allowed_extensions to '*' to permit any extension to be specified by an end-user. [[GH-12847](https://github.com/hashicorp/vault/pull/12847)]
 
 FEATURES:
 
+* **Customizable HTTP Headers**: Add support to define custom HTTP headers for root path (`/`) and also on API endpoints (`/v1/*`) [[GH-12485](https://github.com/hashicorp/vault/pull/12485)]
+* **Elasticsearch in the UI**: Elasticsearch DB is now supported by the UI [[GH-12672](https://github.com/hashicorp/vault/pull/12672)]
```
Comment (briankassouf) [inline CHANGELOG.md:17]: Can we make it more clear this is inside the database secrets engine?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,93 +1,175 @@
...
+* **KV Custom Metadata**: Add ability in kv-v2 to specify version-agnostic custom key metadata via the
+  metadata endpoint. The data will be present in responses made to the data endpoint independent of the
+  calling token's `read` access to the metadata endpoint. [[GH-12907](https://github.com/hashicorp/vault/pull/12907)]
+* **KV patch**: Add partial update support the for the `/<mount>/data/:path` kv-v2
+  endpoint through HTTP `PATCH`.  A new `patch` ACL capability has been added and
+  is required to make such requests. [[GH-12687](https://github.com/hashicorp/vault/pull/12687)]
```
Comment (briankassouf) [inline CHANGELOG.md:23]: Should we call out here that this is in tech preview status?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,93 +1,175 @@
...
 * **Key Management Secrets Engine (Enterprise)**: Adds support for distributing and managing keys in GCP Cloud KMS. [[GH-2158](https://github.com/hashicorp/vault/pull/2158)]
+* **Namespaces (Enterprise)**: Adds support for locking Vault API for particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
```
Comment (briankassouf) [inline CHANGELOG.md:25]:
```suggestion
* **Namespaces (Enterprise)**: Adds support for locking the Vault API for particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,93 +1,175 @@
...
+* **OIDC Authorization Code Flow**: The Vault UI now supports OIDC Authorization Code Flow [[GH-12800](https://github.com/hashicorp/vault/pull/12800)]
```
Comment (briankassouf) [inline CHANGELOG.md:26]: Is this the same feature as the one below?

File: CHANGELOG.md (hunk 1 above)
Comment (kalafut) [reply CHANGELOG.md:17]: Any suggestions on structuring the message? There are a few that are like this.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,93 +1,175 @@
...
+* **OIDC Identity Provider (Tech Preview)**: Adds support for Vault to be an OpenID Connect (OIDC) provider. [[GH-12932](https://github.com/hashicorp/vault/pull/12932)]
+* **Oracle DB in the UI**: Oracle DB connection is now supported in the UI [[GH-12752](https://github.com/hashicorp/vault/pull/12752)]
```
Comment (briankassouf) [inline CHANGELOG.md:28]: Same comment here as with elastic search and postgres below

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,93 +1,175 @@
...
+  allowed_extensions to '*' to permit any extension to be specified by an end-user. [[GH-12847](https://github.com/hashicorp/vault/pull/12847)]
 
 FEATURES:
 
+* **Customizable HTTP Headers**: Add support to define custom HTTP headers for root path (`/`) and also on API endpoints (`/v1/*`) [[GH-12485](https://github.com/hashicorp/vault/pull/12485)]
+* **Elasticsearch in the UI**: Elasticsearch DB is now supported by the UI [[GH-12672](https://github.com/hashicorp/vault/pull/12672)]
+* **KV Custom Metadata**: Add ability in kv-v2 to specify version-agnostic custom key metadata via the
+  metadata endpoint. The data will be present in responses made to the data endpoint independent of the
+  calling token's `read` access to the metadata endpoint. [[GH-12907](https://github.com/hashicorp/vault/pull/12907)]
+* **KV patch**: Add partial update support the for the `/<mount>/data/:path` kv-v2
```
Comment (ncabatoff) [inline CHANGELOG.md:21]:
```suggestion
* **KV patch**: Add partial update support for the `/<mount>/data/:path` kv-v2
```

File: CHANGELOG.md (hunk 2 above)
Comment (kalafut) [reply CHANGELOG.md:23]: Updated

File: CHANGELOG.md (hunk 3 above)
Comment (ncabatoff) [inline CHANGELOG.md]:
```suggestion
* **Namespace Locking (Enterprise)**: Adds support for preventing requests against particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,93 +1,175 @@
...
 * **Key Management Secrets Engine (Enterprise)**: Adds support for distributing and managing keys in GCP Cloud KMS. [[GH-2158](https://github.com/hashicorp/vault/pull/2158)]
+* **Namespaces (Enterprise)**: Adds support for locking Vault API for particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
+* **OIDC Authorization Code Flow**: The Vault UI now supports OIDC Authorization Code Flow [[GH-12800](https://github.com/hashicorp/vault/pull/12800)]
+* **OIDC Identity Provider (Tech Preview)**: Adds support for Vault to be an OpenID Connect (OIDC) provider. [[GH-12932](https://github.com/hashicorp/vault/pull/12932)]
+* **Oracle DB in the UI**: Oracle DB connection is now supported in the UI [[GH-12752](https://github.com/hashicorp/vault/pull/12752)]
+* **Postgres in the UI**: Postgres DB is now supported by the UI [[GH-12945](https://github.com/hashicorp/vault/pull/12945)]
+* **Transform (Enterprise)**: Add advanced features for encoding and decoding for Transform FPE
+* **Transform (Enterprise)**: Add a `reference` field to batch items, and propogate it to the response
```
Comment (briankassouf) [inline CHANGELOG.md:31]: Feels like an improvement vs a feature

File: CHANGELOG.md (hunk 4 above)
Comment (kalafut) [reply CHANGELOG.md:26]: Yeah I'll remove this one and keep the other

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,93 +1,175 @@
...
+* Add ClientID to Token With Entities in Activity Log: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]
```
Comment (ncabatoff) [inline CHANGELOG.md:32]:
```suggestion
* Deduplicate Token With Entities in Activity Log: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]
```

File: CHANGELOG.md (hunk 3 above)
Comment (kalafut) [reply CHANGELOG.md]: Updated

File: CHANGELOG.md (hunk 7 above)
Comment (kalafut) [reply CHANGELOG.md]: Moved

File: CHANGELOG.md (hunk 8 above)
Comment (kalafut) [reply CHANGELOG.md]: Updated

File: CHANGELOG.md (hunk 1 above)
Comment (kalafut) [reply CHANGELOG.md]: Updated in https://github.com/hashicorp/vault/pull/13067/commits/9cdb4f9f01da0fb623ae845f3295964eb5768e78. WDYT?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,93 +1,174 @@
...
+  is required to make such requests. [[GH-12687](https://github.com/hashicorp/vault/pull/12687)]
 * **Key Management Secrets Engine (Enterprise)**: Adds support for distributing and managing keys in GCP Cloud KMS. [[GH-2158](https://github.com/hashicorp/vault/pull/2158)]
+* **Namespace Locking (Enterprise)**: Adds support for preventing requests against particular namespaces. [[GH-2213](https://github.com/hashicorp/vault/pull/2213)]
+* **OIDC Identity Provider (Tech Preview)**: Adds support for Vault to be an OpenID Connect (OIDC) provider. [[GH-12932](https://github.com/hashicorp/vault/pull/12932)]
+* **Oracle Database UI**: The UI now supports adding and editing Oracle connections in the database secret engine [[GH-12752](https://github.com/hashicorp/vault/pull/12752)]
+* **Postgres Database UI**: The UI now supports adding and editing Postgres connections in the database secret engine [[GH-12945](https://github.com/hashicorp/vault/pull/12945)]
+* **Deduplicate Token With Entities in Activity Log**: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]
```
Comment (ncabatoff) [inline CHANGELOG.md:29]:
```suggestion
* **Deduplicate Token Without Entities in Activity Log**: Vault tokens without entities are now tracked with client IDs and deduplicated in the Activity Log [[GH-12820](https://github.com/hashicorp/vault/pull/12820)]
```

**Review by ncabatoff — APPROVED** [review]

**Pull request merged**